scripts-dev/check_newsfragment.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+#
+# A script which checks that an appropriate news file has been added on this
+# branch.
+#
+# As first argument, it requires the PR number, so that it can check that the
+# newsfragment has the correct name.
+#
+# Usage:
+#   ./scripts-dev/check_newsfragment.sh 382
+#
+# Exit codes:
+#   0: all is well
+#   1: the newsfragment is wrong in some way
+#   9: the script has not been invoked properly
+
+echo -e "+++ \033[32mChecking newsfragment\033[m"
+
+set -e
+
+if [ -z "$1" ]; then
+  echo "Please specify the PR number as the first argument (e.g. 382)."
+  exit 9
+fi
+
+
+pull_request_number="$1"
+
+# Print a link to the contributing guide if the user makes a mistake
+CONTRIBUTING_GUIDE_TEXT="!! Please see the contributing guide for help writing your changelog entry:
+https://github.com/matrix-org/sygnal/blob/main/CONTRIBUTING.md#changelog"
```

Comment:
Are we using Sygnal's contributions guidelines on purpose? Shouldn't we copy them over to Sydent instead?

Reply:
Copying them wouldn't make sense, and didn't want this to get blocked behind writing new ones

Reply:
I hadn't noticed this. I think it might be better just not to write anything than to link to the CONTRIBUTING from another project.

Reply:
For what it's worth, in the [PR template](https://github.com/matrix-org/sydent/blob/main/.github/PULL_REQUEST_TEMPLATE.md) we link to Synapse's :D.

Reply:
I've put up a PR (#393) which includes enough of the contributing guide to include the changelog entry guide. (I think it's fairly important to have a reference for it, because I doubt I would personally know what to do if I didn't already know from Synapse/Sygnal).